fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -987,7 +1052,56 @@ public ParseNode visitDecimalValue(StarRocksParser.DecimalValueContext context)
 
     @Override
     public ParseNode visitString(StarRocksParser.StringContext context) {
-        return new StringLiteral(unquote(context.getText()));
+        String quotedString;
+        if (context.SINGLE_QUOTED_TEXT() != null) {
+            quotedString = context.SINGLE_QUOTED_TEXT().getText();
+        } else {
+            quotedString = context.DOUBLE_QUOTED_TEXT().getText();
+        }
+
+        return new StringLiteral(escapeBackSlash(quotedString.substring(1, quotedString.length() - 1)));
+    }
+
+    private static String escapeBackSlash(String str) {
```

Comment:
1. It should be called `unescapeBackSlash`?  
2. How about replace it with a regex?

Reply:
This code is copied from the code in the cup file. The logic of this code is to remove the first
\, so there is no problem calling escapeBackSlash. This PR does not want to modify the logic here, keeping the old code completely consistent and reducing the modification of logic, the function of this PR is already complicated enough. There is a chance to modify it later.